src/main/java/org/truffleruby/core/queue/QueueNodes.java

```diff
@@ -103,8 +93,38 @@ private Object doPop(UnsizedQueue queue) {
             return getContext().getThreadManager().runUntilResult(this, queue::take);
         }
 
+        @Specialization(guards = "!nonBlocking")
+        protected Object popBlocking(RubyQueue self, boolean nonBlocking, long timeoutMilliseconds,
+                @Exclusive @Cached BranchProfile closedProfile) {
+            final UnsizedQueue queue = self.queue;
+            final long timeLimit = System.currentTimeMillis() + timeoutMilliseconds;
```

Comment:
I think a common name for this is `deadline`

Reply:
Renamed.